universe/Universe.cpp

```diff
@@ -2209,77 +2209,82 @@ void Universe::ExecuteEffects(const Effect::TargetsCauses& targets_causes,
 
             if (effects_group != last_effects_group) {
                 last_effects_group = effects_group;
-                dispatched_targets_causes.push_back(std::make_pair(effects_group, Effect::TargetsCauses()));
-                group_targets_causes = &dispatched_targets_causes.back().second;
+                dispatched_targets_causes[effects_group->Priority()].push_back(std::make_pair(effects_group, Effect::TargetsCauses()));
+                group_targets_causes = &dispatched_targets_causes[effects_group->Priority()].back().second;
             }
             group_targets_causes->push_back(*targets_it);
         }
     }
 
     // execute each effects group one by one
-    std::vector<std::pair<Effect::EffectsGroup*, Effect::TargetsCauses> >::iterator effect_group_it;
-    for (effect_group_it = dispatched_targets_causes.begin();
-         effect_group_it != dispatched_targets_causes.end(); ++effect_group_it)
+    std::map<int, std::vector<std::pair<Effect::EffectsGroup*, Effect::TargetsCauses> > >::reverse_iterator priority_group_it;
+    for (priority_group_it = dispatched_targets_causes.rbegin();
```

Comment:
Before I saw the reverse iteration, I posted some comments saying I thought the ordering was lowest first and that that was the wrong way around. "lower priority" suggests a lower number to me. If it was "order" or "rank" or similar then lower first would make more sense, I think...

"least effect overall" is pretty much meaningless in this context... depends on what the effect does and what earlier / later effects did/will do to determine how ordering affects that.